addons/product/views/product_document_views.xml

```diff
@@ -58,9 +60,13 @@
                     </t>
                     <t t-name="kanban-box">
                         <div class="oe_kanban_global_area o_kanban_attachment">
+                            <field name="res_model" invisible="True"/>
                             <div class="ribbon ribbon-top-right" invisible="active">
                                 <span class="text-bg-danger">Archived</span>
                             </div>
+                            <div class="ribbon ribbon-top-right" invisible="not active or res_model == 'product.template'">
```

Comment:
Same here. It slightly disturbs that to set something as a variant, you are looking if it's a template, instead of checking if it's a variant. Personal preference though.